src/main/java/org/jabref/gui/groups/GroupTreeView.java

```diff
@@ -397,6 +415,21 @@ private void setupClearButtonField(CustomTextField customTextField) {
         }
     }
 
+    private void setNewGroupButtonStyle(TreeTableView<GroupNodeViewModel> groupTree) {
+        if (groupTree.getRoot() != null) {
+            if (groupTree.getExpandedItemCount() <= 10) {
+                addNewGroup.setStyle("-fx-border-width: 0px; -fx-background-color: -jr-theme;" +
+                        " -fx-padding: 0.5em; -fx-text-fill: -jr-white;");
+            } else {
+                addNewGroup.setStyle("-fx-border-width: 0px; -fx-background-color: -jr-icon-background-active;" +
+                        " -fx-padding: 0.5em; -fx-text-fill: -jr-black;");
+            }
+        } else {
+            addNewGroup.setStyle("-fx-border-width: 0px; -fx-background-color: -jr-theme;" +
+                    " -fx-padding: 0.5em; -fx-text-fill: -jr-white;");
+        }
+    }
```

Comment:
Instead of hardcoding the css in the java file, please make this a pseudoclass in css, which should be very easy to implement. See `BindingsHelper::includePseudoClassWhen` !

Reply:
Hi @calixtus , i've checked this `BindingsHelper::includePseudoClassWhen` needs a parameter of `node` and the css changes should be applied to the nodes whenever a condition is applied. however, in my change, what I'm changing is the color of `Add Group` button whenever the user added 10 groups (doesn't include the count of subgroups) and not the nodes. Any advise is greatly appreciated.